crypto/BRCryptoPeer.c

```diff
@@ -0,0 +1,100 @@
+//
+//  BRCryptoPeer.h
+//  BRCore
+//
+//  Created by Ed Gamble on 10/17/2019.
+//  Copyright © 2019 breadwallet. All rights reserved.
+//
+//  See the LICENSE file at the project root for license information.
+//  See the CONTRIBUTORS file at the project root for a list of contributors.
+
+#include "BRCryptoPeer.h"
+#include <arpa/inet.h>
+
+/** Forward Declarations */
+static void
+cryptoPeerRelease (BRCryptoPeer peer);
+
+struct BRCryptoPeerRecord {
+    BRCryptoNetwork network;
+    char *address;
+    uint16_t port;
+    char *publicKey;
+
+    struct in_addr inetAddr;
+
+    BRCryptoRef ref;
+};
+
+IMPLEMENT_CRYPTO_GIVE_TAKE (BRCryptoPeer, cryptoPeer);
+
+private_extern BRCryptoPeer
+cryptoPeerCreate (BRCryptoNetwork network,
+                  const char *address,
+                  uint16_t port,
+                  const char *publicKey) {
+    // Require `address`
+    if (NULL == address || 0 == strlen (address)) return NULL;
+
+    // Require `address` to parse as a INET address
+    struct in_addr addr;
+    if (-1 == ascii2addr (AF_INET, address, &addr)) return NULL;
```

Comment:
How does IPv6 (which seems to be supported by the `BRPeerManager`) factor?

Reply:
Also, not sure `ascii2addr` is portable.

Reply:
CORE-578